src/module_1/module_1_meteo_api.py

```diff
@@ -1,7 +1,136 @@
-""" This is a dummy example """
+import requests
+import pandas as pd
+import matplotlib.pyplot as plt
+from typing import Dict, Optional, List
+
+API_URL = "https://climate-api.open-meteo.com/v1/climate?"
+
+COORDINATES = {
+    "Madrid": {"latitude": 40.4165, "longitude": -3.7026},
+    "London": {"latitude": 51.507351, "longitude": -0.127758},
+    "Rio": {"latitude": -22.906847, "longitude": -43.172896},
+}
+VARIABLES = "temperature_2m_mean,precipitation_sum,soil_moisture_0_to_10cm_mean"
+
+def make_api_request(url: str, params: Dict[str, str])-> Optional[Dict]:
+    """
+    Generic function to make an API request and extract data in JSON format.
+    :param url: API URL
```

Comment:
I know there are some people that use it but I don't find that adding parameters like this in docstrings adds anything but verbosity to the code. If good naming and typed parameter definitions is used, I find that most of the times it becomes self-explanatory.

I would only add this kind of extra explanation when it's not clear or it adds value